src/validators/arguments.rs

```diff
@@ -278,28 +318,55 @@ impl Validator for ArgumentsValidator {
+                                    Ok(value) => {
+                                        output_kwargs
+                                            .set_item(either_str.as_py_string(py, state.cache_str()), value)?;
+                                    }
+                                    Err(ValError::LineErrors(line_errors)) => {
+                                        for err in line_errors {
+                                            errors.push(err.with_outer_location(raw_key.clone()));
+                                        }
+                                    }
+                                    Err(err) => return Err(err),
+                                },
+                                None => {
+                                    if let ExtraBehavior::Forbid = self.extra {
+                                        errors.push(ValLineError::new_with_loc(
+                                            ErrorTypeDefaults::UnexpectedKeywordArgument,
+                                            value,
+                                            raw_key.clone(),
+                                        ));
                                     }
                                 }
-                                Err(err) => return Err(err),
                             },
-                            None => {
-                                if let ExtraBehavior::Forbid = self.extra {
-                                    errors.push(ValLineError::new_with_loc(
-                                        ErrorTypeDefaults::UnexpectedKeywordArgument,
-                                        value,
-                                        raw_key.clone(),
-                                    ));
-                                }
+                            VarKwargsMode::UnpackedTypedDict => {
+                                // Save to the remaining kwargs, we will validate as a single dict:
+                                remaining_kwargs.set_item(either_str.as_py_string(py, state.cache_str()), value)?;
+                            }
+                        }
+                    }
+                }
+
+                if self.var_kwargs_mode == VarKwargsMode::UnpackedTypedDict {
+                    // `var_kwargs_validator` is guaranteed to be `Some`:
+                    match self
+                        .var_kwargs_validator
+                        .as_ref()
+                        .unwrap()
+                        .validate(py, remaining_kwargs.as_any(), state)
+                    {
+                        Ok(value) => {
+                            output_kwargs.update(value.downcast_bound::<PyDict>(py).unwrap().as_mapping())?;
+                        }
+                        Err(ValError::LineErrors(line_errors)) => {
+                            for error in line_errors {
+                                errors.push(error);
```

Comment:
I've tried using one of the `extend*` methods but they expect clonable elements

Reply:
I think `errors.extend(line_errors)` should work fine (moves elements rather than cloning).